src/test/scala/vsys/blockchain/state/contract/vswap/ExecuteVSwapValidDiffTest.scala

```diff
@@ -0,0 +1,340 @@
+    ExecuteContractFunctionTransaction, ExecuteContractFunctionTransaction, ExecuteContractFunctionTransaction, ExecuteContractFunctionTransaction)] = for {
+    // register 3 tokens and VSwap contract, then deposit all 3 tokens into contract
+    // the function will issue and deposit the entire supply of liquidity tokens
+    (genesis, genesis2, master, user, regTokenA, regTokenB, regLiquidity, regVSwap, issueTokenA, issueTokenB, issueLiquidity, depositA, depositB, depositLiquidity, fee, ts, attach) <-
+      createABLiquidityTokenAndInitVSwap(
+      1000, // total supply of token A
+      1, // unity of token A
+      1000, // issue amount of token A
+      1000, // total supply of token B
+      1, // unity of token B
+      1000, // issue amount of token B
+      1000, // total supply of liquidity token
+      1000, // unity of liquidity token
+      10, // minimum liquidity
+      10, // deposit amount of token A
+      10) // deposit amount of token B
+    //withdraw tokens
+    withdrawTokenA <- withdrawToken(master, regTokenA.contractId, regVSwap.contractId.bytes.arr, master.toAddress.bytes.arr, 10L, fee, ts + 10)
+    withdrawTokenB <- withdrawToken(master, regTokenB.contractId, regVSwap.contractId.bytes.arr, master.toAddress.bytes.arr, 10L, fee, ts + 11)
+    withdrawLiquidity <- withdrawToken(master, regLiquidity.contractId, regVSwap.contractId.bytes.arr, master.toAddress.bytes.arr, 10L, fee, ts + 12)
+  } yield (genesis, genesis2, regTokenA, regTokenB, regLiquidity, regVSwap, issueTokenA, issueTokenB, issueLiquidity, depositA, depositB, depositLiquidity,
+    withdrawTokenA, withdrawTokenB, withdrawLiquidity)
+
+  property("vswap able to withdraw balance") {
+    forAll(preconditionsAndVSwapDepositAndWithdrawLiquidityTokens) { case (genesis: GenesisTransaction, genesis2: GenesisTransaction, regTokenA: RegisterContractTransaction,
+    regTokenB: RegisterContractTransaction, regLiquidity: RegisterContractTransaction, regVSwap: RegisterContractTransaction, issueTokenA: ExecuteContractFunctionTransaction,
+    issueTokenB: ExecuteContractFunctionTransaction, issueLiquidity: ExecuteContractFunctionTransaction, depositA: ExecuteContractFunctionTransaction, depositB: ExecuteContractFunctionTransaction,
+    depositLiquidity: ExecuteContractFunctionTransaction, withdrawA: ExecuteContractFunctionTransaction, withdrawB: ExecuteContractFunctionTransaction,
+    withdrawLiquidity: ExecuteContractFunctionTransaction) =>
+      assertDiffAndStateCorrectBlockTime(Seq(TestBlock.create(genesis.timestamp, Seq(genesis, genesis2)), TestBlock.create(regTokenA.timestamp, Seq(regTokenA,
+        regTokenB, regLiquidity, regVSwap, issueTokenA, issueTokenB, issueLiquidity, depositA, depositB, depositLiquidity))),
+        TestBlock.createWithTxStatus(withdrawLiquidity.timestamp, Seq(withdrawLiquidity), TransactionStatus.Success)) { (blockDiff, newState) =>
+        blockDiff.txsDiff.txStatus shouldBe TransactionStatus.Success
```

Comment:
Why only Status.Success? where is the detailed state check? balance update? state var update, state map update?